src/App.tsx

```diff
@@ -1,26 +1,139 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import * as todosService from './api/todos';
+import { Todo } from './types/Todo';
+import { ErrorsEnum } from './utils/ErrorsEnum';
+import { Error } from './components/Error/Error';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todoList, setTodosList] = useState<Todo[]>([]);
+  const [filteredTodoList, setFilteredTodoList] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [error, setError] = useState<ErrorsEnum | null>(null);
+  const [isEnabled, setIsEnabled] = useState(true);
+  const [isClearDisabled, setIsClearDisabled] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+
+  const updateClearDisabled = useCallback(() => {
+    setIsClearDisabled(!todoList.some(todo => todo.completed));
+  }, [todoList]);
+
+  useEffect(() => {
+    todosService
+      .getTodos()
+      .then(todos => {
+        setTodosList(todos);
+        setFilteredTodoList(todos);
+        updateClearDisabled();
+      })
+      .catch(() => {
+        setError(ErrorsEnum.UNABLE_LOAD_TODOS);
+        setTimeout(() => setError(null), 3000);
+      });
+  });
+
+  const handleDeleteTodo = (todoId: number) => {
+    setIsEnabled(false);
+    todosService
+      .deleteTodos(todoId)
+      .then(() => {
+        setTodosList(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+        setFilteredTodoList(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+      })
+      .catch(() => {
+        setError(ErrorsEnum.UANBLE_DELETE_TODO);
+        setTimeout(() => setError(null), 3000);
```

Comment:
This _setTimeOut_ could be used inside a an _useEffect_ hook that tracks _error_.
Why? For now, you only have this error, but in the future, you can have multiple errors being fired everywhere. (one for deleting a Todo, another for creating one, etc...)
Using a hook to track the error and apply the timeOut will shorten your code and keep all appearances of this timeout in a single place